From 4a5ecbd5ee145df8a83db6f2c4074ab4d5c3b72c Mon Sep 17 00:00:00 2001
From: Dmitry Petrochenko <dmitry.petrochenko@intel.com>
Date: Thu, 21 Mar 2013 20:02:51 +0700
Subject: Dalvik: Fix a typo in assert (writeBackVR)

BZ: 94860

The typo caused crashes of assert build.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I41400733e3a56af54d10c94e107c922d87d6ff57
Orig-MCG-Change-Id: I5df78cc3785276a7b8461b9fe0989fa9bb4198bd
Signed-off-by: Dmitry Petrochenko <dmitry.petrochenko@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/codegen/x86/AnalysisO1.cpp |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/vm/compiler/codegen/x86/AnalysisO1.cpp b/vm/compiler/codegen/x86/AnalysisO1.cpp
index 93e1c44..9c8d569 100644
--- a/vm/compiler/codegen/x86/AnalysisO1.cpp
+++ b/vm/compiler/codegen/x86/AnalysisO1.cpp
@@ -2634,9 +2634,9 @@ void writeBackVR(int vR, LowOpndRegType type, int physicalReg) {
     // Paranoid check because we only handle writing back if in either
     // GP or XMM registers
     assert((physicalReg >= PhysicalReg_StartOfGPMarker &&
-            physicalReg <= PhysicalReg_StartOfGPMarker) ||
+            physicalReg <= PhysicalReg_EndOfGPMarker) ||
             (physicalReg >= PhysicalReg_StartOfXmmMarker &&
-                    physicalReg <= PhysicalReg_StartOfXmmMarker));
+                    physicalReg <= PhysicalReg_EndOfXmmMarker));
 
     // If VR is already in memory, we can skip writing it back
     if (isInMemory(vR, getRegSize(physicalType))) {
-- 
1.7.4.1

